### ISSUE TITLE
error message: Unsupported publisher event type.

### ISSUE THREAD

**Issue by ZhenshengLee** [continental/rmw_ecal#16]
## Environment

- ubuntu2004
- ros2 foxy

## Action

```sh
export RMW_IMPLEMENTATION=rmw_ecal_dynamic_cpp
ros2 run demo_nodes_cpp talker
```

## Behavior

```sh
>>> [rcutils|error_handling.c:108] rcutils_set_error_state()
This error state is being overwritten:

  'Handle's typesupport identifier (rosidl_typesupport_c) is not supported by this library
, at /tmp/binarydeb/ros-foxy-rosidl-typesupport-c-1.0.2/src/type_support_dispatch.hpp:116'

with this new error message:

  'Unsupported publisher event type., at /home/zs/zs_ws/ga2_ws/src/ga_ros2/ga_deps/rmw_ecal/rmw_ecal_dynamic_cpp/src/event.cpp:37'

rcutils_reset_error() should be called after error handling to avoid this.
```

## Comment

May be we should comment this error msg in event.cpp:37?

**Comment by ZhenshengLee** [comment]
Solved in v1.0.

**Comment by ZhenshengLee** [comment]
This error msg still exist in recent version.

```
'Unsupported publisher event type., at /home/zs/zs_ws/ga2_ws/src/ga_ros2/ga_deps/rmw_ecal/rmw_ecal_shared_cpp/src/event.cpp:39'
```
